ground/src/main/java/com/google/android/ground/model/geometry/Polygon.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.model.geometry
+
+import com.google.common.collect.ImmutableList
+
+class PolygonExteriorHoleException(val shellMaximum: Coordinate, val holeMaximum: Coordinate) :
+    GeometryException("Invalid polygon. Holes must be contained within a polygon's shell, but the maximum coordinates of the shell $shellMaximum are lesser than the maximum coordinates of a hole $holeMaximum")
```

Comment:
Can this be split to multiple lines and then use trimIndent() to remove the extra whitespace from the final string. It ends up being much more readable.

```
"""
    Example of a long string
    that is written over multiple lines
""".trimIndent()

Reply:
sgtm, thanks for the suggestion!